### PR TITLE
Define commands globally

### PR DESCRIPTION
**Pull request by rdnetto**
This PR moves the definition of the commands, so that they are available in all buffers, instead of just the Haskell buffers.

This is desirable so that we can, for example, reload Intero while editing .cabal files. It also eliminates some error messages that were shown if a non-Haskell file was open when existing, since InteroKill would not be  defined in the active buffer.

### CODE REVIEW

File: after/ftplugin/haskell/intero.vim [hunk 1]
```diff
@@ -3,10 +3,6 @@ if exists('b:did_ftplugin_intero') && b:did_ftplugin_intero
 endif
 let b:did_ftplugin_intero = 1
 
-if !has('patch-7.4.1578')
```
Comment (parsonsmatt) [inline after/ftplugin/haskell/intero.vim]: Why is this removed?

**Comment by parsonsmatt** [comment]
Thanks @rdnetto ! I'm curious why the check for the `timer_start` patch level was removed.

File: after/ftplugin/haskell/intero.vim (hunk 1 above)
Comment (rdnetto) [inline after/ftplugin/haskell/intero.vim]: It's been moved into `plugin/intero.vim`. In hindsight, it makes sense to have it in both, so I'll re-add it.

Comment (parsonsmatt) [inline after/ftplugin/haskell/intero.vim]: :+1:

**Comment by parsonsmatt** [comment]
@rdnetto I'm going to give you commit access to the repository. Feel free to merge whenever you're happy with the PR.

**Comment by rdnetto** [comment]
Cheers. :)
